jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/assertions/DataPointAttributes.java

```diff
@@ -40,21 +42,30 @@ public static AttributeMatcher attributeWithAnyValue(String name) {
   }
 
   /**
-   * Create a set of attribute matchers that should be used to verify set of data point attributes.
+   * Create a group of attribute matchers that should be used to verify list of data point
+   * attributes.
    *
-   * @param attributes list of matchers to create set. It must contain matchers with unique names.
-   * @return set of unique attribute matchers
+   * @param attributes list of matchers to create group from. It must contain matchers with unique
+   *     names.
+   * @return list of unique attribute matchers
    * @throws IllegalArgumentException if provided list contains two or more matchers with the same
    *     name.
-   * @see MetricAssert#hasDataPointsWithAttributes(Set[]) for detailed description of the algorithm
-   *     used for matching
+   * @see MetricAssert#hasDataPointsWithAttributes(Collection[]) for detailed description of the
+   *     algorithm used for matching
    */
-  public static Set<AttributeMatcher> attributeSet(AttributeMatcher... attributes) {
-    Set<AttributeMatcher> matcherSet = Arrays.stream(attributes).collect(Collectors.toSet());
-    if (matcherSet.size() < attributes.length) {
-      throw new IllegalArgumentException(
-          "Duplicated matchers found in " + Arrays.toString(attributes));
+  public static List<AttributeMatcher> attributeSet(AttributeMatcher... attributes) {
```

Comment:
What do you think about renaming this method to attributeGroup ? The current one is somehow suggesting that Set is returned